src/network.ts

```diff
@@ -54,5 +55,56 @@ export const albHttpToHttpsRedirection: ResourceValidationPolicy = {
                 return;
             }
         }),
-    };
+        validateResourceOfType(aws.alb.Listener, (listener, _, reportViolation) => {
+            if (listener.protocol !== "HTTP") {
+                return;
+            }
+            // Not a compliance problem per-say, but certainly odd enough to report.
+            if (!listener.defaultActions || listener.defaultActions.length === 0) {
+                reportViolation(`HTTP listener has no default actions configured.`);
+                return;
+            }
+            if (listener.defaultActions.length > 1) {
+                reportViolation(`HTTP listener has more than one default action.`);
+                return;
+            }
+            const defaultAction = listener.defaultActions[0];
+            const compliant = true
+                && defaultAction.type === "redirect"
+                && defaultAction.redirect
+                && defaultAction.redirect.protocol === "HTTPS";
+
+            if (!compliant) {
+                const msg = `Default action for HTTP listener must be a redirect using HTTPS.`;
+                reportViolation(msg);
+                return;
+            }
+        }),
+        validateResourceOfType(aws.lb.Listener, (listener, _, reportViolation) => {
+            if (listener.protocol !== "HTTP") {
+                return;
+            }
+            // Not a compliance problem per-say, but certainly odd enough to report.
+            if (!listener.defaultActions || listener.defaultActions.length === 0) {
+                reportViolation(`HTTP listener has no default actions configured.`);
+                return;
+            }
+            if (listener.defaultActions.length > 1) {
+                reportViolation(`HTTP listener has more than one default action.`);
```

Comment:
If the user recieves only this message, I don't think it conveys the reason that more than one default action is a problem. Perhaps just adding a little more like the following would help?

```suggestion
                reportViolation(`HTTP listener has more than one default action. HTTP listener should only be used to redirect to HTTPS`);
```